grpc/core/src/main/java/io/helidon/grpc/core/CollectingObserver.java

```diff
@@ -25,69 +25,69 @@
 import io.grpc.stub.StreamObserver;
 
 /**
- * A {@link StreamObserver}.
+ * Utility {@link StreamObserver} mostly used for testing.
  *
- * @param <T> ToDo: Add JavaDoc
- * @param <V> ToDo: Add JavaDoc
- * @param <U> ToDo: Add JavaDoc
- * @param <A> ToDo: Add JavaDoc
- * @param <R> ToDo: Add JavaDoc
+ * @param <T> the type of input elements to the reduction operation
+ * @param <A> the mutable accumulation type of the reduction operation
+ * @param <R> the result type of the reduction operation
+ * @param <U> the type of values observed
+ * @param <V> the request type before conversion
  */
 public class CollectingObserver<T, V, U, A, R> implements StreamObserver<V> {
+
     private final Collector<T, A, R> collector;
     private final StreamObserver<U> responseObserver;
     private final Function<V, T> requestConverter;
     private final Function<R, U> responseConverter;
     private final Consumer<Throwable> errorHandler;
-
     private final A accumulator;
 
     /**
-     * ToDo: Add JavaDoc.
+     * Constructor.
      *
-     * @param collector        ToDo: Add JavaDoc
-     * @param responseObserver ToDo: Add JavaDoc
+     * @param collector the collector
+     * @param observer the observer
      */
-    public CollectingObserver(Collector<T, A, R> collector, StreamObserver<U> responseObserver) {
-        this(collector, responseObserver, null, null, null);
+    public CollectingObserver(Collector<T, A, R> collector, StreamObserver<U> observer) {
+        this(collector, observer, null, null, null);
```

Comment:
This needs to be refactored:
- we do not support `null` as a parameter of a public executable, either methods are needed for each combination, or use a builder
- we do not use public constructors except for type where it is a common practice (Exceptions, InputStream etc. - not sure if this applies), use static `create` factor methods.